src/lightmap.cpp

```diff
@@ -492,6 +492,46 @@ bool map::pl_line_of_sight( const tripoint &t, const int max_range ) const
 
 #include "messages.h"
 
+typedef ShadowcastingSlope slope;
+
+slope::ShadowcastingSlope( int x, int y )
+{
+    // Changing x1/y1 < x2/y2 to x1*y2 < x2*y1 assumes that y1 and y2 have the same sign.
+    // To do this, we can remove the sign from y and apply it to x.
+    ydiff = abs( y );
+    if( y < 0 ) {
+        xdiff = -x;
+    } else {
+        xdiff = x;
+    }
+}
+
+inline bool operator< ( const slope &lhs, const slope &rhs )
+{
+    // x1/y1 < x2/y2 === x1*y2 < x2*y1 if y1 and y2 have the same sign.
+    return lhs.xdiff * rhs.ydiff < rhs.xdiff * lhs.ydiff;
+}
+inline bool operator> ( const slope &lhs, const slope &rhs )
+{
+    return rhs < lhs;
```

Comment:
Uh... because that would just make it call itself until it fills up the call stack?